scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -711,6 +721,10 @@
            old-file-server-state (:sandbox-file-server-container-state old-state)]
        (when (and (= new-file-server-state :running) (not= old-file-server-state :running))
          (record-sandbox-url pod-name new-state)))
+     ; Hostname will change for pods scheduled by Kubernetes.
+     (when-not (= (api/pod->node-name (:pod new-state))
```

Comment:
Should this be gated on "twosigma.com/scheduler" "kubernetes"? (FYI: This is code that's live immediately)

Reply:
Done, good idea. Moved the gate to this clause to make it extra unlikely to affect live code.